pkg/importer/http-datasource.go

```diff
@@ -116,6 +116,18 @@ func NewHTTPDataSource(endpoint, accessKey, secKey, certDir string, contentType
 func (hs *HTTPDataSource) Info() (ProcessingPhase, error) {
 	var err error
 	hs.readers, err = NewFormatReaders(hs.httpReader, hs.contentLength)
+	hs.url, _ = url.Parse(fmt.Sprintf("nbd:unix:%s", nbdkitSocket))
```

Comment:
Can we move
```go
if hs.contentType == cdiv1.DataVolumeArchive {
  return ProcessingPhaseTransferDataDir, nil
}
```
To line 119. I keep thinking the err gets overwritten (it doesn't, but it is hard to read). But if the readers error I don't think there is a need to start nbdkit at all.

Reply:
Done, also rebased